resource.go

```diff
@@ -26,8 +26,13 @@ type ResourceConfig struct {
 	ClusterEndpointFunc func(v interface{}) (string, error)
 	ClusterIDFunc       func(v interface{}) (string, error)
 	ClusterStatusFunc   func(v interface{}) (providerv1alpha1.StatusCluster, error)
-	Logger              micrologger.Logger
-	NodeCountFunc       func(v interface{}) (int, error)
+	// TODO replace this with a G8sClient to fetch the node versions from the
```

Comment:
@xh3b4sd I saw you posted this TODO, still valid?

Reply:
I think meanwhile we are moving into a direction where `statusresource` just not be a thing anymore in the future. Let's not do too much in here.